src/main/java/frc/robot/subsystems/Drivetrain.java

```diff
@@ -154,53 +157,67 @@ public void initSpeedMode()
         rightFrontMotor.config_kI(1, SPEED_I_CONSTANT, TIMEOUT_MS);
         rightFrontMotor.config_kD(1, SPEED_D_CONSTANT, TIMEOUT_MS);
         rightFrontMotor.config_kF(1, SPEED_F_CONSTANT, TIMEOUT_MS);
-        
-	}
+
+    }
+
     public void setControlMode(ControlMode mode) {
- 		leftFrontMotor.set(mode, 0);
+        leftFrontMotor.set(mode, 0);
         rightFrontMotor.set(mode, 0);
     }
-	 public ControlMode getControlMode() {
+
+    public ControlMode getControlMode() {
         return leftFrontMotor.getControlMode();
     }
-   
-	 public void setPercentSpeedPID (double setSpeedL, double setSpeedR)
-    {
+
+    public void setPercentSpeedPID(double setSpeedL, double setSpeedR) {
         setSpeedR = MathUtil.clamp(setSpeedR, -1, 1);
-		    setSpeedL = MathUtil.clamp(setSpeedL, -1, 1);
+        setSpeedL = MathUtil.clamp(setSpeedL, -1, 1);
         leftFrontMotor.set(ControlMode.Velocity, MAX_TICKS_PER_SECOND * setSpeedL);
         rightFrontMotor.set(ControlMode.Velocity, MAX_TICKS_PER_SECOND * setSpeedR);
     }
-       
-	public boolean getReversed(){
+
+    public boolean getReversed() {
         return isReversed;
     }
-    public void setReversed(boolean r){
+
+    public boolean getMultiplierSpeed() {
+        return speedMultiplierActive;
+    }
+
+    public void setReversed(boolean r) {
         isReversed = r;
     }
-	 public void writePIDs(double output){
-		leftFrontMotor.pidWrite(output);
-		rightFrontMotor.pidWrite(-output);
- }
+
+    public void setMultiplierSpeed(boolean active, double multiplier) {
+        if(multiplier == speedMultiplier || !speedMultiplierActive)     //this mode is only toggled if: this mode is off already OR the same button that turned this mode on was pressed again
```

Comment:
Something I'd like to start doing is using Java's doc comments. You can see examples of them in `resetOdometry`. What you write will appear when you hover over the name of the function, so that's where I'd describe what the behavior of those if statements are. It's a good behavior, but it's confusing if it's not stated beforehand